File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> result = new ArrayList();
        if (root == null) return result;
        List<TreeNode> currents = new ArrayList();
        currents.add(root);
        while(currents.size() > 0) {
            int size = currents.size();
            Integer rightSideVal = null;
            for (int i = 0; i < size; i++) {
                TreeNode node = currents.remove(0);
                if (node.left != null) currents.add(node.left);
                if (node.right != null) currents.add(node.right);
                rightSideVal = node.val;
            }
            result.add(rightSideVal);
        }
        return result;
    }

}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun rightSideView(root: TreeNode?): List<Int> {
        val result = mutableListOf<Int>()
        if (root == null) return result
        val currents = mutableListOf<TreeNode>()
        currents.add(root)
        while (currents.isNotEmpty()) {
            val size = currents.size
            for (i in 0..size-1) {
                val node = currents.removeAt(0)
                if (i == size - 1) {
                    result.add(node.`val`)
                }
                if (node.left != null) currents.add(node.left)
                if (node.right != null) currents.add(node.right)
            }
        }
        return result
    }
}